Tests/SwiftSyntaxTest/DiagnosticTest.swift

```diff
@@ -18,9 +18,16 @@ extension Diagnostic.Message {
   /// Suggestion for the user to explicitly check a value does not equal zero.
   static let checkEqualToZero =
     Diagnostic.Message(.note, "check for explicit equality to '0'")
+
+  static func badFunction(_ name: TokenSyntax) -> Diagnostic.Message {
+    return .init(.error, "bad function '\(name.text)'")
+  }
+  static func endOfFunction(_ name: TokenSyntax) -> Diagnostic.Message {
+    return .init(.warning, "end of function '\(name.text)'")
+  }
 }
 
-public class DiagnosticsTestCase: XCTestCase {
+public class DiagnosticTestCase: XCTestCase {
```

Comment:
Rename it to `DiagnosticTestCase.swift`?

Right now it looks like _all_ of the test files and the test classes inside them are named inconsistently; it might be cleaner to do a separate PR that unifies them all (and do it on both branches) instead of trying to add it to this cherrypick.

Reply:
That makes sense. Let's tackle it in another PR then.